mapiproxy/libmapiproxy/dcesrv_mapiproxy_session.c

```diff
@@ -40,63 +44,54 @@
    \return Pointer to an allocated mpm_session structure on success,
    otherwise NULL
  */
-struct mpm_session *mpm_session_new(TALLOC_CTX *mem_ctx, 
-				     struct server_id serverid,
-				     uint32_t context_id)
+struct mpm_session *mpm_session_new(struct server_id serverid,
+				    uint32_t context_id,
+				    const char *username,
+				    struct GUID	*uuid)
 {
 	struct mpm_session	*session = NULL;
 
-	if (!mem_ctx) return NULL;
-
-	session = talloc_zero(mem_ctx, struct mpm_session);
+	session = talloc_zero(NULL, struct mpm_session);
 	if (!session) return NULL;
 
 	session->server_id = serverid;
 	session->context_id = context_id;
-	session->ref_count = 0;
+	if (uuid) session->uuid = *uuid;
 	session->destructor = NULL;
 	session->private_data = NULL;
 
-	return session;
-}
-
-
-/**
-   \details Increment the ref_count associated to a session
-
-   \param session pointer to the session where to increment ref_count
-
-   \return true on success, otherwise false
- */
-bool mpm_session_increment_ref_count(struct mpm_session *session)
-{
-	if (!session) return false;
+	// Released RPC connection, session may be kept to avoid premature release
+	// of private_data, that will be cleared when all user sessions are released
+	session->released = false;
+	session->username = talloc_strdup(session, username);
+	if (!session->username) return NULL;
 
-	session->ref_count += 1;
+	DLIST_ADD_END(mpm_sessions, session, struct mpm_session *);
```

Comment:
Any reason inserting at the end? Personally I'd rather prefer `DLIST_ADD` because the `DLIST_ADD_END` macro has a third unused parameter. Not very important though.